pkg/controllers/rollout/controller_test.go

```diff
@@ -162,12 +163,7 @@ func TestReconcilerHandleResourceSnapshot(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			queue := &controllertest.Queue{TypedInterface: workqueue.NewTypedRateLimitingQueue[reconcile.Request](workqueue.DefaultTypedItemBasedRateLimiter[reconcile.Request]())}
 			handleResourceSnapshot(tt.snapshot, queue)
-			if tt.shouldEnqueue && queue.Len() == 0 {
-				t.Errorf("handleResourceSnapshot test `%s` didn't queue the object when it should enqueue", name)
-			}
-			if !tt.shouldEnqueue && queue.Len() != 0 {
-				t.Errorf("handleResourceSnapshot test `%s` queue the object when it should not enqueue", name)
-			}
+			validateEnqueueBehavior(t, name, queue, tt.shouldEnqueue, nil)
```

Comment:
need to update existing ut, just validating if shouldEnqueue is not good enough for now. We need to check the key.

Reply:
Updated all tests.